python/tskit/tables.py

```diff
@@ -259,12 +260,17 @@ def metadata_schema(self, metadata_schema):
     def metadata_schema(self):
         self.ll_table.metadata_schema = None
 
+    def validate_and_encode_metadata_row(self, metadata):
+        if self.metadata_schema is None:
+            return metadata
+        else:
+            validate(metadata, self.metadata_schema["schema"])
```

Comment:
Be nice to define catch jsonschema's exception here and raise our own. We don't want jsonschema to become part of our API --- maybe we'll want to ditch it for something better later.

Reply:
See exceptions.ProvenanceValidationError, eg.

Reply:
Done in https://github.com/tskit-dev/tskit/pull/491/commits/38de515732f6e037b237e0bb0d6808ae9ca33ce6